packages/notifications/src/inAppMessaging/providers/pinpoint/utils/helpers.ts

```diff
@@ -252,8 +258,15 @@ export const extractContent = ({
 				PrimaryBtn,
 				SecondaryBtn,
 			} = content;
-			const defaultPrimaryButton = PrimaryBtn?.DefaultConfig;
-			const defaultSecondaryButton = SecondaryBtn?.DefaultConfig;
+
+			const getButtonConfig = (button?: InAppMessageButton) => ({
+				...button?.DefaultConfig,
+				...button?.[configPlatform],
+			});
+
+			const defaultPrimaryButton = PrimaryBtn ? getButtonConfig(PrimaryBtn) : undefined;
+			const defaultSecondaryButton = SecondaryBtn ? getButtonConfig(SecondaryBtn) : undefined;
```

Comment:
Nit: Related to the above comment, the responsibility for checking existence becomes solely the function's
```suggestion
			const defaultPrimaryButton = getButtonConfig(PrimaryBtn);
			const defaultSecondaryButton = getButtonConfig(SecondaryBtn);
```

Reply:
updated here:
[extracted getButtonConfig fn and refactored to add data existence che…](https://github.com/aws-amplify/amplify-js/pull/12917/commits/7ff8b96827cccda8f1a14598f8f426652cfbb879)